### PR TITLE
[eas-cli] Add build:list and build:view commands

### PR DESCRIPTION
**Pull request by satya164**
This commit adds build:list and build:view commands to show status of builds.

When running build:list, it shows last 10 builds, but it can be configured by passing status, platform and limit filters.

When running build:view, it shows the last build. An id param can be passed to view a specific build.

Screen Shot 2020-12-09 at 13 40 06
Screen Shot 2020-12-09 at 13 41 48
Screen Shot 2020-12-09 at 13 42 25
Screen Shot 2020-12-09 at 13 42 51
Screen Shot 2020-12-09 at 13 43 49
Screen Shot 2020-12-09 at 13 44 08

### CODE REVIEW

File: packages/eas-cli/package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "strip-ansi": "^6.0.0",
     "tar": "^6.0.5",
     "tempy": "^0.7.0",
+    "terminal-link": "^2.1.1",
```
Comment (wkozyra95) [inline packages/eas-cli/package.json:53]: Please do not use terminal-link, It does not work correctly for some terminals  https://github.com/expo/eas-cli/issues/89

File: packages/eas-cli/src/commands/build/view.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const projectId = await getProjectIdAsync(projectDir);
+    const accountName = await getProjectAccountNameAsync(projectDir);
+
+    const spinner = ora().start('Fetching the build…');
+
+    try {
+      let build: Build;
+
+      if (buildId == null) {
```
Comment (wkozyra95) [inline packages/eas-cli/src/commands/build/view.ts:32]: I think that for this command buildId should required.

File: packages/eas-cli/src/commands/build/view.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  static args = [{ name: 'buildId' }];
+
+  async run() {
+    const { buildId } = this.parse(BuildList).args;
+
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const projectId = await getProjectIdAsync(projectDir);
+    const accountName = await getProjectAccountNameAsync(projectDir);
+
+    const spinner = ora().start('Fetching the build…');
```
Comment (wkozyra95) [inline packages/eas-cli/src/commands/build/view.ts:27]: I'm not sure about that, I think that commands like this should have as little output as possible. Only the actual data.

Comment (satya164) [reply packages/eas-cli/src/commands/build/view.ts:30]: It's doing a network request, so without a spinner, it'll look like it's stuck, especially on slower connections

Comment (satya164) [reply packages/eas-cli/src/commands/build/view.ts:30]: @wkozyra95 It's doing a network request, so without a spinner, it'll look like it's stuck, especially on slower connections

File: packages/eas-cli/src/commands/build/view.ts (hunk 2 above)
Comment (satya164) [reply packages/eas-cli/src/commands/build/view.ts:35]: @wkozyra95 The design doc in notion mentions that we should show last build if no id is specified

**Review by wkozyra95 — APPROVED** [review]

File: packages/eas-cli/src/build/utils/formatBuild.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        const url = build.artifacts?.buildUrl;
+
+        if (!url) {
+          return chalk.red('not found');
+        }
+
+        return url;
+      },
+    },
+    { label: 'Started at', get: build => new Date(build.createdAt).toLocaleString() },
```
[the remark this answers is not in the document]
Comment (brentvatne) [reply packages/eas-cli/src/build/utils/formatBuild.ts:61]: we use "Start time" on the web UI, we could use the same wording potentially

File: packages/eas-cli/src/commands/build/list.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    try {
+      const response = await apiClient.get<{ data: { builds: Build[] } }>(
+        `projects/${projectId}/builds`,
+        {
+          searchParams: { platform, status, limit },
+          responseType: 'json',
+        }
+      );
+
+      const builds = response.body.data.builds;
```
Comment (dsokal) [inline packages/eas-cli/src/commands/build/list.ts:45]:
```suggestion
      const { builds } = response.body.data;
```

File: packages/eas-cli/src/commands/build/view.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  static args = [{ name: 'buildId' }];
+
+  async run() {
+    const { buildId } = this.parse(BuildView).args;
+
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const projectId = await getProjectIdAsync(projectDir);
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectName = `@${accountName}/${exp.slug}`;
```
Comment (dsokal) [inline packages/eas-cli/src/commands/build/view.ts:28]: There's a similar if not the same block of code for `build:list`. Maybe it'd be worth creating a util for this?

File: packages/eas-cli/src/build/utils/formatBuild.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      label: 'Platform',
+      get: build => {
+        switch (build.platform) {
+          case 'ios':
+            return 'iOS';
+          case 'android':
+            return 'Android';
+          default:
+            return 'Unknown';
+        }
```
Comment (dsokal) [inline packages/eas-cli/src/build/utils/formatBuild.ts:23]: Use `platformDisplayNames` from `constants.ts` instead.

**Review by dsokal — APPROVED** [review]

File: packages/eas-cli/src/build/utils/formatBuild.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (satya164) [reply packages/eas-cli/src/build/utils/formatBuild.ts:61]: We'd also need to use "Finish time", which imo sounds kinda werider than "Finished at"

**Pull request merged**